syntax = "proto3";

package inventoryorder;

message Item {
  string id = 1;
  string name = 2;
  int32 quantity = 3;
}

message StockRequest {
  string item_id = 1;
  int32 quantity = 2;
}

message StockResponse {
  bool available = 1;
  string message = 2;
}

message Ack {
  string status = 1;
  string message = 2;
}

message OrderRequest {
  string order_id = 1;
  string item_id = 2;
  int32 quantity = 3;
}

message OrderResponse {
  string status = 1;
  string message = 2;
}

service InventoryService {
  rpc AddItem (Item) returns (Ack);
  rpc CheckStock (StockRequest) returns (StockResponse);
  rpc DeductStock (StockRequest) returns (Ack);
}
